Scripts/Sources/Automation/Shell.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+class Shell {
+    static func executeAndPrint(_ command: String, timeout: Double = 10, allowFailure: Bool = false) throws {
+        print("$ \(command)")
+        let output = try executeShell(commandPath: "/bin/bash", arguments: ["-c", command], timeout: timeout, allowFailure: allowFailure) {
+            print($0, separator: "", terminator: "")
+        }
+        if let error = output?.error {
+            print(error)
+        }
+    }
+
+    static func execute(_ command: String, timeout: Double = 10, allowFailure: Bool = false) throws -> (output: String?, error: String?)? {
+        return try executeShell(commandPath: "/bin/bash", arguments: ["-c", command], timeout: timeout, allowFailure: allowFailure)
+    }
+
+    static func bash(commandName: String,
+                     arguments: [String] = [],
+                     timeout: Double = 10,
+                     allowFailure: Bool = false) throws -> (output: String?, error: String?)? {
+        guard let execution = try? executeShell(commandPath: "/bin/bash" ,
+                                                arguments: [ "-l", "-c", "/usr/bin/which \(commandName)" ],
+                                                timeout: 1),
+            var whichPathForCommand = execution?.output else { return nil }
+
+        whichPathForCommand = whichPathForCommand.trimmingCharacters(in: NSCharacterSet.whitespacesAndNewlines)
+        return try executeShell(commandPath: whichPathForCommand, arguments: arguments, timeout: timeout, allowFailure: allowFailure)
+    }
+
+    static func executeShell(commandPath: String,
+                             arguments: [String] = [],
+                             timeout: Double = 10,
+                             allowFailure: Bool = false,
+                             stream: @escaping (String) -> Void = { _ in }) throws -> (output: String?, error: String?)? {
+        let task = Process()
+        task.launchPath = commandPath
+        task.arguments = arguments
+
+        let pipeForOutput = Pipe()
+        task.standardOutput = pipeForOutput
+
+        let pipeForError = Pipe()
+        task.standardError = pipeForError
+        task.launch()
+
+        let fileHandle = pipeForOutput.fileHandleForReading
+        fileHandle.waitForDataInBackgroundAndNotify()
+
+        var outputData = Data()
+
+        func process(data: Data) {
+            outputData.append(data)
+            if let output = String(data: data, encoding: .utf8) {
+                stream(output)
+            }
+        }
+
+        let observer = NotificationCenter.default.addObserver(forName: Notification.Name.NSFileHandleDataAvailable, object: fileHandle, queue: nil) { notification in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/tevelee/Eval/blob/fc973589729700ac4c55183b359a3f3047c6157f/Scripts/Sources/Automation/Shell.swift#L59"></span>Line should be 120 characters or less: currently 165 characters</td>
    </tr>
  </tbody>
</table>